source/server/listener_manager_impl.cc

```diff
@@ -1018,6 +1019,11 @@ Network::DrainableFilterChainSharedPtr ListenerFilterChainFactoryBuilder::buildF
 
   auto& config_factory = Config::Utility::getAndCheckFactory<
       Server::Configuration::DownstreamTransportSocketConfigFactory>(transport_socket);
+  if (transport_socket.name() == Extensions::TransportSockets::TransportSocketNames::get().Quic &&
+      typeid(Quic::QuicServerTransportSocketConfigFactory) != typeid(config_factory)) {
+    throw EnvoyException(fmt::format("error building filter chain for quic listener: wrong "
+                                     "transport socket specified for quic transport socket."));
```

Comment:
think it's worth a bit more detail, saying which is the correct transport socket, and or dumping the relevant listener config?

Reply:
Extended the error message by suggesting the correct config to use and dumping the wrong config.